lm_eval/models/huggingface.py

```diff
@@ -705,17 +703,16 @@ def tok_batch_encode(
         old_padding_side = self.tokenizer.padding_side
         self.tokenizer.padding_side = padding_side
 
+        add_special_tokens = {}
         if self.AUTO_MODEL_CLASS == transformers.AutoModelForCausalLM:
-            add_special_tokens = False or self.add_bos_token
-        elif self.AUTO_MODEL_CLASS == transformers.AutoModelForSeq2SeqLM:
-            add_special_tokens = True
+            add_special_tokens = {"add_special_tokens": False or self.add_bos_token}
```

Comment:
why the change in how we define `add_special_tokens`? This doesn't change functionality but is a bit confusing IMO to read